### PR TITLE
Poetry docker

### PR DESCRIPTION
**Pull request by Kleczyk**
Add new poetry files, docker-compose.yml, dockerfile, and make scripts for new people and everything  is decrypted  in README . database without changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,21 +1,147 @@
-📚 research_LLM 📚
+# Polish benchmark LLM
+Polish benchmark LLM is a compact yet powerful platform designed for benchmarking  in polish langue open-source large language models (LLMs)  Equipped with diverse test datasets, it facilitates rigorous and comprehensive evaluation of LLMs. This tool streamlines the process of assessing model performance, enabling researchers and developers to reliably test and share their findings in the dynamic landscape of AI and natural language processing.
```
Comment (finloop) [inline README.md:2]: **nitpick(non-bocking):** Add newline

```suggestion
# Polish benchmark LLM

Polish benchmark LLM is a compact yet powerful platform designed for benchmarking  in polish langue open-source large language models (LLMs)  Equipped with diverse test datasets, it facilitates rigorous and comprehensive evaluation of LLMs. This tool streamlines the process of assessing model performance, enabling researchers and developers to reliably test and share their findings in the dynamic landscape of AI and natural language processing.
```

Comment (Kleczyk) [reply README.md]: done

File: README.md [hunk 2]
```diff
@@ -1,21 +1,148 @@
-📚 research_LLM 📚
+# Polish benchmark LLM
 
-Project Objective 🎯
-The aim of the "research_LLM" project is to conduct tests and assessments of various Language Models (LLMs - Large Language Models) to understand their performance and applications.
+Polish benchmark LLM is a compact yet powerful platform designed for benchmarking  in polish langue open-source large language models (LLMs)  Equipped with diverse test datasets, it facilitates rigorous and comprehensive evaluation of LLMs. This tool streamlines the process of assessing model performance, enabling researchers and developers to reliably test and share their findings in the dynamic landscape of AI and natural language processing.
```
Comment (finloop) [inline README.md:3]: **nitpick(non-blocking):** I would rephrase "Polish benchmark LLM" as "Polish LLM benchmark"

File: README.md [hunk 3]
```diff
@@ -1,21 +1,148 @@
...
+
+# Running the Benchmark
+
+### For Linux
+
+**First Time Setup**
+
+Build and start the Docker containers. This step is required only once or after any changes in the Docker configuration.
+```bash 
+sudo docker-compose up --build
```
Comment (finloop) [inline README.md:72]: **issue(blocking):** Use `docker compose` instead of `docker-compose`. 

More on this subject here: https://stackoverflow.com/questions/66514436/difference-between-docker-compose-and-docker-compose

Comment (finloop) [inline README.md:72]: **issue(blocking):** Remove sudo from all commands using docker. Linux users should follow official docker guide, which shows how to use docker without sudo.

File: README.md [hunk 4]
```diff
@@ -1,21 +1,148 @@
...
+```
+
+**Regular Usage**
+
+Start the Docker containers without rebuilding.
+```bash 
+sudo docker-compose up
```
Comment (finloop) [inline README.md:79]: **issue(blocking):** Remove sudo from all commands using docker. Linux users should follow official docker guide, which shows how to use docker without sudo.

File: README.md [hunk 5]
```diff
@@ -1,21 +1,148 @@
...
+``` 
+
+**Shutdown**
+
+To stop and remove the containers, networks, and volumes, use:
+```bash 
+sudo docker-compose down
+```
```
Comment (finloop) [inline README.md:87]: **issue(blocking):** This willl NOT remove ANY volumes. To remove all named volumes use: `-v` flag.

Eg:
```sh
docker compose down -v
``` 

Note: This can cause data loss!

Docs: https://docs.docker.com/engine/reference/commandline/compose_down/

File: README.md [hunk 6]
```diff
@@ -1,21 +1,148 @@
...
+### For Linux Users
+
+Clone the repository to get a local copy on your machine,
+Change into the project directory,
+Run the setup script to configure the environment.
+
+   ```bash
+   git clone https://github.com/knmlprz/Benchmark_LLM.git
+   cd Benchmark_LLM
+   ./trust_me_bro.sh
```
Comment (finloop) [inline README.md:46]: **issue(blocking)**: Non-descriptive name of file.

File: benchmark/main.py [hunk 7]
```diff
@@ -0,0 +1 @@
+print("hello world")
```
Comment (finloop) [inline benchmark/main.py:1]: **issue(blocking):** ?? what is this for?

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+version: '3.8'
+
+services:
+  app:
+    build: .
+    volumes:
+      - ./benchmark:/usr/src/app
```
Comment (finloop) [inline docker-compose.yml:7]: **issue(blocking):** Why you're mounting this volume? Your code is already in the container (see Dockerfile):

Line 14:
```Dockerfile
COPY . .
```

File: dockerfile [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+
+COPY pyproject.toml ./
+
+# Install poetry for package management
+RUN pip install --no-cache-dir poetry
+RUN poetry install
+
+
+COPY . .
```
Comment (finloop) [inline dockerfile:14]: **suggesion(non-blocking):** Do all files need to be copied into container? You can use `.dockerignore` to exclude some. See: https://docs.docker.com/build/building/context/#dockerignore-files

File: dockerfile [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+
+WORKDIR /usr/src/app
+
+
+COPY pyproject.toml ./
+
+# Install poetry for package management
+RUN pip install --no-cache-dir poetry
+RUN poetry install
```
Comment (finloop) [inline dockerfile:11]: **suggestion(non-blocking):** See this article on how to build a good dockerfile for poetry (includes caching etc.): https://medium.com/@albertazzir/blazing-fast-python-docker-builds-with-poetry-a78a66f5aed0

File: pyproject.toml [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+authors = ["Daniel Kleczynski <danielkleczynski@gmial.com>"]
+
+[tool.poetry.dependencies]
+python = "^3.11"
+python-dotenv = "^1.0.0"
+aiohttp = {extras = ["speedups"], version = "^3.9.0"}
+pydantic = "^2.5.1"
+langchain = "^0.0.329"
+openai = "^1.0.0rc3"
+pysqlite3 = "^0.5.0"
```
Comment (finloop) [inline pyproject.toml:14]: **suggestion(non-blocking):** Python has a built-in support for sqlite3: https://docs.python.org/3/library/sqlite3.html

Comment (finloop) [inline pyproject.toml:14]: **issue(blocking):** Unused dependencies

File: trust_me_bro.sh [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+# Installs Poetry
+pip install poetry
+
+echo "Poetry has been installed."
+
+# Install dependencies using Poetry
+poetry install
+
+echo "Dependencies installed."
```
Comment (finloop) [inline trust_me_bro.sh:23]: **suggestion(non-blocking):** Run this script through a tool like shellcheck: https://www.shellcheck.net/

This will catch errors like:
Your script does not exit on error. If one of your commands eg. `python3 -m venv venv` returns non-zero status, your script will still try to execute next lines:
```
pip install --upgrade pip
source venv/bin/activate <- this will fail
pip install poetry <- this will install poetry somewhere (depends on your setup) but NOT in venv
```

Comment (finloop) [reply trust_me_bro.sh:23]: https://ioflood.com/blog/bash-exit-on-error/

File: README.md (hunk 3 above)
Comment (Kleczyk) [reply README.md:72]: I don't know about it :heart:

**Comment by finloop** [comment]
Use formatter for .md (https://stackoverflow.com/questions/48339757/vscode-how-to-format-markdown-file) and .py (black) files.

File: README.md [hunk 13]
```diff
@@ -1,21 +1,151 @@
-📚 research_LLM 📚
+<<<<<<< HEAD
+# Polish LLM benchmark
+=======
+# Polish benchmark LLM
+>>>>>>> 466a40031a73a16c8a419573fa9461786ad3043c
```
Comment (finloop) [inline README.md:5]: **issue(blocking):** Unresolved conflict

File: README.md [hunk 14]
```diff
@@ -1,21 +1,151 @@
...
 
-Project Stages 📊
+### [Getting Started](#getting-started)
+- [For Liunx](#for-linux-users)
+- [For Windows](#for-Windows-users)
 
-- Model Selection 🧐
-        Selecting different language models for testing.
+### [Run Benchamrk](#run-benchmark)
+- [For Liunx](#for-linux)
```
Comment (finloop) [inline README.md:15]: **nitpick(non-blocking):** No newline between header and list.

File: benchmark/main.py
Comment (finloop) [inline benchmark/main.py:1]: **suggesion(non-blocking):** Remove empty file

File: benchmark/model.py
Comment (finloop) [inline benchmark/model.py:1]: **suggesion(non-blocking):** Remove empty file

File: set_venv.sh [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+
+            # Install dependencies using Poetry
+            poetry install.
+    else
+        echo "Virtual environment failed to activate."
+    fi
+    
+else
+    echo "Virtual environment already exists."
+fi
```
Comment (finloop) [inline set_venv.sh:35]: **issue(blocking):** Make bash exit on error. Don't use ifs.

Read this: ->>>>>> https://ioflood.com/blog/bash-exit-on-error/

**Review by finloop — CHANGES_REQUESTED** [review]
**suggestion**: Reformat all files (mdlint, black)

**Review by finloop — CHANGES_REQUESTED** [review]
See above.

**Review by finloop — APPROVED** [review]
LGTM
